Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonFilter.cc

```diff
@@ -0,0 +1,126 @@
+// system include files
+#include <atomic>
+#include <memory>
+#include <cmath>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDFilter.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/HcalCalibObjects/interface/HcalHBHEMuonVariables.h"
+
+//#define EDM_ML_DEBUG
+//
+// class declaration
+//
+
+namespace alcaHcalHBHEMuon {
+  struct Counters {
+    Counters() : nAll_(0), nGood_(0), nFinal_(0) {}
+    mutable std::atomic<unsigned int> nAll_, nGood_, nFinal_;
+  };
+}  // namespace alcaHcalHBHEMuon
+
+class AlCaHcalHBHEMuonFilter : public edm::global::EDFilter<edm::RunCache<alcaHcalHBHEMuon::Counters> > {
+public:
+  AlCaHcalHBHEMuonFilter(edm::ParameterSet const&);
+
+  std::shared_ptr<alcaHcalHBHEMuon::Counters> globalBeginRun(edm::Run const&, edm::EventSetup const&) const override;
+
+  bool filter(edm::StreamID, edm::Event&, edm::EventSetup const&) const override;
+  void globalEndRun(edm::Run const& iRun, edm::EventSetup const&) const override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  // ----------member data ---------------------------
+  int prescale_;
+  edm::InputTag labelHBHEMuonVar_;
+  edm::EDGetTokenT<HcalHBHEMuonVariablesCollection> tokHBHEMuonVar_;
+};
+
+//
+// constructors and destructor
+//
+AlCaHcalHBHEMuonFilter::AlCaHcalHBHEMuonFilter(edm::ParameterSet const& iConfig)
+    : prescale_(iConfig.getParameter<int>("prescale")),
+      labelHBHEMuonVar_(iConfig.getParameter<edm::InputTag>("hbheMuonLabel")),
+      tokHBHEMuonVar_(consumes<HcalHBHEMuonVariablesCollection>(labelHBHEMuonVar_)) {
+  edm::LogVerbatim("HBHEMuon") << "Parameters read from config file \n\t prescale_ " << prescale_ << "\n\t Labels "
+                               << labelHBHEMuonVar_;
+}  // AlCaHcalHBHEMuonFilter::AlCaHcalHBHEMuonFilter  constructor
+
```

Comment:
isn't the destructor missing here?

Reply:
No, the destructor is defined in
https://github.com/cms-sw/cmssw/blob/76ab0ba0ab44a1c8ee28be749d3db8aade2c987d/Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonFilter.cc#L35

Reply:
Added. I was relying on the default one which gets automatically added

Reply:
@makortel I think we looked at the file at a different point in time, now it's indeed there